main.py

```diff
@@ -0,0 +1,60 @@
+from typing import List
+
+from fastapi import FastAPI, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+import crud
+import schemas
+from db.engine import SessionLocal
+
+app = FastAPI()
+
+
+def get_db() -> Session:
+    db = SessionLocal()
+
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+@app.get("/")
+async def root():
+    return {"message": "Hello World"}
+
+
+@app.get("/authors/", response_model=List[schemas.Author])
+def get_authors(db: Session = Depends(get_db)):
+    return crud.get_authors(db)
+
+
+@app.post("/authors/", response_model=schemas.Author)
+def create_author(
+    author: schemas.AuthorBaseCreate,
+    db: Session = Depends(get_db),
+):
+    db_author = crud.get_author_by_name(db=db, name=author.name)
+
+    if db_author is None:
+        raise HTTPException(status_code=404, detail="Author not found")
+
+    return crud.create_author(db=db, author=author)
+
+
+@app.get("/books/", response_model=List[schemas.Book])
+def get_books(db: Session = Depends(get_db)):
+    return crud.get_books(db)
```

Comment:
The `get_books` endpoint should support pagination by accepting `skip` and `limit` query parameters to retrieve a subset of books. Consider adding these parameters and using them in the query.